### PR TITLE
Improved handling of config settings.

### PR DESCRIPTION
**Pull request by dimitri-yatsenko**
* `dj.config` now has methods `save_local` and `save_global` for saving the settings explicitly.
* Removed support for vagrant and for readthedocs.
* Updated CHANGELOG.md

### CODE REVIEW

File: datajoint/admin.py [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-    if 'yes' == user_choice('Do you wish to change the password?'):
-        connection = conn() if connection is None else connection
-        connection.query("SET PASSWORD = PASSWORD('%s')" % new_password)
-        print('Done!')
-    else:
-        print('Password left unchanged')
+def set_password(new_password=None, connection=None):   # pragma: no cover
+    connection = conn() if connection is None else connection
+    if new_password is None:
+        new_password = getpass()
```
Comment (eywalker) [inline datajoint/admin.py]: A prompt should be added here.

File: datajoint/settings.py [hunk 2]
```diff
@@ -93,6 +93,18 @@ def __iter__(self):
...
+        """
+        saves the settings in the local config file  
+        """
+        self.save(LOCALCONFIG)
+
+    def save_global(self):
+        """
+        saves the settings in the global config file  
+        """
+        self.save(os.path.expanduser('~/' + GLOBALCONFIG))
```
Comment (eywalker) [inline datajoint/settings.py]: To be safe, I recommend that we use `path.join` from `os.path` package when combining paths.

File: datajoint/admin.py (hunk 1 above)
Comment (dimitri-yatsenko) [inline datajoint/admin.py]: `getpass` shows a prompt.

File: datajoint/settings.py (hunk 2 above)
Comment (dimitri-yatsenko) [inline datajoint/settings.py]: Does `expanduser` work for Windows?  Sorry i don't have a windows system.

File: datajoint/admin.py (hunk 1 above)
Comment (eywalker) [inline datajoint/admin.py]: Ah ok cool, didn't know that it had default prompt.

File: datajoint/settings.py (hunk 2 above)
Comment (eywalker) [inline datajoint/settings.py]: Yes, it will default to `HOMEPATH`, so at least it is consistently defined.

**Pull request merged**